### ISSUE TITLE
Platform.ready() called before cordova deviceready for livereload builds (this.platform.is("cordova") also returns false).

### ISSUE THREAD

**Issue by sachingarg05** [ionic-team/angular-toolkit#436]
### Bug:
platform ready() is called 'before' cordova deviceready when doing a livereload build. And it is called with readySource = "dom" instead of "cordova".

### Reason:
ionic:platform code to setup ready() promise runs before cordova.js has loaded. So it decides that it is not in cordova environment and doesn't need to wait for deviceready.

### ~~Why this seems to be happening:~~ 
(Probably not due to this, see my comment below)

When doing `ionic cordova run ios --livereload`, cordova.js is added in script tags **at the end**. This could be the reason why vendor.js/main.js code doesn't find window.cordova object when settings up platform.ready promise.
![image](https://user-images.githubusercontent.com/4737179/104112039-c5053600-530f-11eb-93ed-62420f53eea6.png)
Same behavior as above when trying to run using:
`ionic serve --no-open --external --consolelogs --engine=cordova --platform=ios`

When doing `ionic cordova prepare ios`, cordova.js is added in script tags **before** vendor.js or main.js. So things work OK here.
![image](https://user-images.githubusercontent.com/4737179/104112107-be2af300-5310-11eb-8378-83f7d23c58ff.png)

**Possible Fix**
~~Change order in which code adds cordova.js script tag for livereload/serve builds. It should be before code adds script tags vendor.js or main.js.~~

Ionic Info
```
Ionic:

   Ionic CLI                     : 6.12.1 (/Users/user/.npm-global/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 4.11.13
   @angular-devkit/build-angular : 0.801.3
   @angular-devkit/schematics    : 8.1.3
   @angular/cli                  : 8.1.3
   @ionic/angular-toolkit        : 2.3.0 // Pinned to this slightly older version because https://github.com/ionic-team/angular-toolkit/issues/435

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : android 9.0.0, ios 6.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, (and 30 other plugins)

Utility:

   cordova-res (update available: 0.15.2) : 0.15.1
   native-run (update available: 1.3.0)   : 1.2.2

System:

   Android SDK Tools : 26.1.1 (/Users/user/Library/Android/sdk)
   ios-deploy        : 1.11.3
   ios-sim           : 8.0.2
   NodeJS            : v12.19.0 (/usr/local/bin/node)
   npm               : 6.13.6
   OS                : macOS Catalina
   Xcode             : Xcode 11.6 Build version 11E708
```

**Comment by sachingarg05** [comment]
This doesn't seem to be related to order to script tags as I suggested above.

Manually setting `window.document.addEventListener("deviceready" ` works but even after deviceready `this.platform.is("cordova")` returns false.

**Comment by mhartington** [comment]
@sachingarg05 do you have a sample project that we can debug? Otherwise, we can't debug the issue.

**Comment by sachingarg05** [comment]
Not sure if this is due to Ionic 4 or due to large number of native plugins in our project resulting in a timing-issue/race-condition (30 plugins, app compiles to 125 MB on Android). Will try to create a sample project.

**Comment by mhartington** [comment]
@sachingarg05 Hey, any luck getting a sample project together?

**Comment by sachingarg05** [comment]
We couldn't reproduce it in a fresh ionic5 project, and our main project was on ionic4 AND it didn't use cordova-plugin-ionic-webview (we stayed on file:// when we started using ionic to not have to migrate saved preferences of existing users).

We have now update our app to ionic5 + cordova-plugin-ionic-webview + a small custom plugin that reads old stored prefs from file:// origin and sends it to main cordova+ionic webview. (We couldn't make ionicons5 work with file:// even with whatwg-fetch polyfill + cordova-plugin-wkwebview-file-xhr combination that fixed it for ionic4).

Will try reproducing if this problem still exists with ionic5 in this new setup and report back with a sample project. Some of our plugins are closed-source and if removing them from sample project hides the bug that will complicate it. All this might take a few days.
